### ISSUE TITLE
New Feature: Request a system update

### ISSUE THREAD

**Issue by bytetheapple** [FlyingDiver/Indigo-MyQ#31]
feature request: Please add manual polling of cloud server to get current device state from MyQ cloud

The problem:
I have a garage door device, created with indigodomo MyQ plugin.  I have used Homekit bridge to expose the device to Homekit (Siri).  Because the plugin only checks for status updates from the MyQ cloud server every 10 min, Homekit has a delayed view of the device state.  If I open the door with Siri, Homekit still is being told by Indigo that the state of the door is closed (locked) until the next time the plugin polls the cloud server at MyQ.  My common use case is to open the door with Siri, leave the garage, and try to close the door with Siri.  However, when I try to close the door, HomeKit still thinks it is closed, so does nothing.  I need to wait 10 min after opening for Homekit to recognize that it is open so that my close command will work.

I would like to use a trigger or action  group in Indiogdomo to force a state update after the door open/close command is issued.  However, the plugin does not expose this command so I have no way of forcing a device state update.

**Comment by FlyingDiver** [comment]
System updates use the standard Indigo "Request Status" commands.  Which is a button in the UI for the device, or the "Request Full Status Update" universal controls action.  If you have multiple MyQ devices, requesting an update for any of them will update them all.

 
<img width="959" alt="Screen Shot 2022-12-09 at 7 21 34 AM" src="https://user-images.githubusercontent.com/72976/206701960-b55d5764-cb90-4511-b493-30232c1b01f2.png">

**Comment by FlyingDiver** [comment]
Oh - if you put a tilt sensor on the door, you can link it to the MyQ device and changes in the sensor state will trigger a device update automatically.

**Comment by bytetheapple** [comment]
Thank you for the quick response.  Sorry for my ignorance on this but this should, clearly, do the trick.

Sheldon

**Comment by bytetheapple** [comment]
Hi Joe,

Just a quick follow-up.  Didn’t have time to get back to this problem until now.   Without a tilt sensor, I realized I can solve the problem another way and wanted to share in case it is something you hadn’t considered.

I changed the MyQ app notification settings so that it sends an email alert when the state of the garage door changes.
I have it send the email to an account monitored by the Indigodomo Email+ plugin.
I set a trigger on a Regex for “door opened” etc…
I set the action for the trigger to be “Request Full Status Update” whenever the trigger fires.

Now the device on Indigodomo is updated almost instantaneously and so is Homekit, so Siri knows the current state of the doors in near real-time.

Bit of a Kludge, but since I am only using it to sync Homekit/Siri, it is more than good enough.

Hope someone else can benefit from the “scheme”.

Thanks,

Sheldon
